src/core/projectinfo.cpp

```diff
@@ -508,3 +510,125 @@ void ProjectInfo::restoreSettings( QString &projectFilePath, QgsProject *project
   }
   settings.endGroup();
 }
+
+QVariantMap ProjectInfo::getTitleDecorationConfiguration()
+{
+  QVariantMap configuration;
+  const QString configurationName = QStringLiteral( "TitleLabel" );
+
+  if ( QgsProject::instance()->readBoolEntry( configurationName, QStringLiteral( "/Enabled" ), false ) )
+  {
+    QString text = QgsProject::instance()->readEntry( configurationName, QStringLiteral( "/Label" ), QString() );
+    if ( !text.isEmpty() )
+    {
+      text.replace( QStringLiteral( "\n" ), QStringLiteral( "<br>" ) );
+      QTextDocument doc;
+      doc.setHtml( text );
+      text = doc.toPlainText();
+    }
+
+    QColor backgroundColor = QgsColorUtils::colorFromString( QgsProject::instance()->readEntry( configurationName, QStringLiteral( "/BackgroundColor" ), QStringLiteral( "0,0,0,99" ) ) );
+    QColor color = QColor( Qt::black );
+    QColor outlineColor = QColor( Qt::white );
+    bool hasOutline = false;
+
+    QDomDocument doc;
+    QDomElement elem;
+    const QString textXml = QgsProject::instance()->readEntry( configurationName, QStringLiteral( "/Font" ) );
+    if ( !textXml.isEmpty() )
+    {
+      doc.setContent( textXml );
+      elem = doc.documentElement();
+      QgsReadWriteContext rwContext;
+      rwContext.setPathResolver( QgsProject::instance()->pathResolver() );
+      QgsTextFormat textFormat;
+      textFormat.readXml( elem, rwContext );
+
+      color = textFormat.color();
+      color.setAlphaF( textFormat.opacity() );
+      if ( textFormat.buffer().enabled() )
+      {
+        hasOutline = true;
+        outlineColor = textFormat.buffer().color();
+        outlineColor.setAlphaF( textFormat.buffer().opacity() );
+      }
+    }
+
+    configuration["text"] = text;
+    configuration["backgroundColor"] = backgroundColor;
+    configuration["color"] = color;
+    configuration["hasOutline"] = hasOutline;
+    configuration["outlineColor"] = outlineColor;
```

Comment:
Should we add a small class for the config to avoid all the literals?

Reply:
@domi4484 , I'd rather not at this moment; there's another decoration - logo decoration - which will end up reshaping this a bit, and I'm not sure what settings would overlap and not. Let's go with a variant map configuration for now and revisit if/when we implement that other decoration.